fv3core/decorators.py

```diff
@@ -149,7 +149,96 @@ def __iter__(self):
         return self.cache.__iter__()
 
 
-class FV3StencilObject:
+class StencilWrapper:
+    """Wrapped GT4Py stencil object."""
+
+    def __init__(
+        self,
+        func: Callable[..., None],
+        origin: Optional[Index3D] = None,
```

Comment:
with this implementation we don't really have a benefit any more for initializing with origin and domain. Do we still want to do it nevertheless for readability or should we revert that functionality?

Reply:
I feel like it is good to keep it for explicitness.